gcsfs/core.py

```diff
@@ -837,6 +837,8 @@ async def _mkdir(
         location=None,
         create_parents=True,
         enable_versioning=False,
+        uniform_access=False,
```

Comment:
I wonder how many kwargs we should pass before we start to make some Options dataclass

Reply:
Fair. Would you be in favor of replicating the entire [Buckets:Insert](https://cloud.google.com/storage/docs/json_api/v1/buckets/insert) api? Or at least the [gcloud storage buckets create](https://cloud.google.com/sdk/gcloud/reference/storage/buckets/create) options? I wanted to add these in particular because they are the default (and recommended) options for buckets created through the console UI.

Looks like some others we are missing are class/autoclass, retention policy and requester pays. Plus some more niche options.

Reply:
How about we allow "iamConfiguration" as a kwarg, so the caller can put whatever they like in it? We could also other arguments that the API has - then it's just a JSON blob we pass along.